File: TechModule/Programming Fundamentals/07.ObjectsAndClasses - Exercises/04.AverageGrades/Student.cs

using System.Collections.Generic;
using System.Linq;

namespace _04.AverageGrades
{
    public class Student
    {
        public string Name { get; set; }
        public List<double> Grades { get; set; }
        public double AverageGrade => Grades.Sum() / Grades.Count;
    }
}

File: Java-DB-Fundamentals/DatabasesFrameworks-Hibernate&SpringData/02.OOPOverview/src/P07/Student.java
package P07;


import java.util.List;

public class Student {
    private String name;
    private List<Double> grades;
    private Double averageGrade;

    public Student(String name, List<Double> grades){
        this.name = name;
        this.grades = grades;
    }

    public String getName(){
        return this.name;
    }
    public Double getAverageGrade(){
        return grades.stream().mapToDouble(a->a/grades.size()).sum();
    }
}
